easybuild/easyblocks/m/motioncor2.py

```diff
@@ -0,0 +1,123 @@
+##
+# Copyright 2019-2019 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing MotionCor2, implemented as an easyblock
+
+@author: Ake Sandgren, (HPC2N, Umea University)
+"""
+
+import os
+import stat
+
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import adjust_permissions, copy_file, mkdir, write_file
+from easybuild.tools.modules import get_software_root
+
+
+class EB_MotionCor2(EasyBlock):
+    """
+    Support for installing MotionCor2
+     - creates wrapper that loads the correct version of CUDA before
+     - running the actual binary
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(EB_MotionCor2, self).__init__(*args, **kwargs)
+
+        self.cuda_mod_name, self.cuda_name = None, None
+        self.motioncor2_bin = None
+        for dep in self.cfg.dependencies():
+            if dep['name'] == 'CUDA':
+                self.cuda_mod_name = dep['short_mod_name']
+                self.cuda_name = os.path.dirname(self.cuda_mod_name)
+                cuda_ver = dep['version']
+                cuda_short_ver = "".join(cuda_ver.split('.')[:2])
+                self.motioncor2_bin = 'MotionCor2_%s-Cuda%s' % (self.version, cuda_short_ver)
+                break
+
+    def prepare_step(self, *args, **kwargs):
+        super(EB_MotionCor2, self).prepare_step(*args, **kwargs)
+
+        # This is technically already done in __init__
```

Comment:
@akesandgren Why not do everything in `prepare_step`?

You ran into trouble earlier because the workaround in the tests only "hacks" `get_software_root`, not `.dependencies()`.

This looks a bit messy now, we shouldn't be dancing around limitations in the tests by tweaking an easyblock like this...

Reply:
So self.cfg.dependencies() is still available in prepare_step, should have thought of that..

I'l fix it...